Classes/Middleware/LanguageRedirectionMiddleware.php

```diff
@@ -30,22 +30,32 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
             return $handler->handle($request);
         }
 
-        // Get the iso code of the browser language from HTTP request header
-        $browserLanguageIsoCode = substr($request->getServerParams()['HTTP_ACCEPT_LANGUAGE'], 0, 2);
-
-        foreach ($siteLanguages as $siteLanguage) {
-            // Check if the browser language is supported by comparing two letter iso codes
-            if ($browserLanguageIsoCode === $siteLanguage->getTwoLetterIsoCode()) {
-                // Do nothing, if the site language base URL is the currently requested URL
-                if ((string) $siteLanguage->getBase() === $request->getAttribute('normalizedParams')->getRequestUrl()) {
-                    return $handler->handle($request);
+        // Get the iso codes of the browser languages from HTTP request header
+        $browserLanguageIsoCodes = $this->getBrowserLangugeIsoCodes($request->getServerParams()['HTTP_ACCEPT_LANGUAGE']);
+
+        // Check which of the browser languages are supported by comparing two letter iso codes
+        foreach ($browserLanguageIsoCodes as $browserLanguageIsoCode) {
+            foreach ($siteLanguages as $siteLanguage) {
+                if ($browserLanguageIsoCode === $siteLanguage->getTwoLetterIsoCode()) {
+                    // Do nothing, if the site language base URL is the currently requested URL
+                    if ((string) $siteLanguage->getBase() === $request->getAttribute('normalizedParams')->getRequestUrl()) {
+                        return $handler->handle($request);
+                    }
+                    // Redirect the user to the preferred language base URL
+                    return new RedirectResponse($siteLanguage->getBase());
                 }
-
-                // Redirect the user to the preferred language base URL
-                return new RedirectResponse($siteLanguage->getBase());
             }
         }
 
         return $handler->handle($request);
     }
+
+    protected function getBrowserLangugeIsoCodes($acceptLanguageHeader)
```

Comment:
Please add some type hinting to this method.

Reply:
Resolved on the following comment:

`// This function takes in an Accept-Language header string and returns an array of unique two-letter ISO codes`

Reply:
I think the comment is not necessary, but you can add type hinting in TYPO3 source code like this:

```
protected function getBrowserLangugeIsoCodes(string $acceptLanguageHeader): array
```